src/catalog/src/memory/objects.rs

```diff
@@ -403,6 +405,57 @@ impl UpdateFrom<durable::Cluster> for Cluster {
     }
 }
 
+// TODO(jkosh44) Make this infallible and convert to `From`.
+impl TryFrom<&Cluster> for CreateClusterPlan {
```

Comment:
In general not a fan of `TryFrom` implementations unless you actually need to generically program against types that support a particular conversion. My rule of thumb is that unless there are multiple `From` types for the same `To` type the `TryFrom` implementation isn't doing anything for you. Also makes it hard to pass in types, if it turns out to be necessary to pass in another type (e.g., the catalog) to do the conversion.

Is there a particular reason for this implementation, or could it just be a method named `Cluster::try_to_plan`?

Reply:
> Is there a particular reason for this implementation, or could it just be a method named Cluster::try_to_plan?

No particular reason, those are all good points, I'll switch to `try_to_plan`.